e2e/func-e_run_test.go

```diff
@@ -145,10 +142,12 @@ func envoyRunTest(t *testing.T, test func(context.Context, *funcE, *adminClient)
 	_, err = process.NewProcessWithContext(ctx, c.envoyPid) // because os.FindProcess is no-op in Linux!
 	require.Error(t, err, "expected func-e to terminate Envoy after running [%v]", c)
 
-	return c, func() {
+	t.Cleanup(func() {
 		// this may not be present if the process was kill -9'd so don't error
-		os.Remove(c.runDir + ".tar.gz") //nolint
-	}
+		require.NoError(t, os.Remove(c.runDir+".tar.gz"))
```

Comment:
this isn't a test condition, rather a cleanup.. if we did require here and it failed.. wouldn't we mask another assertion failure?

Reply:
ex. the comment `// this may not be present if the process was kill -9'd so don't error` was saying why we didn't enforce errors here, so if we do enforce them now, we should delete that comment. I think it is better to leave it as it wasn't that simple to figure this out the first time 😅